boa/src/builtins/typed_array/integer_indexed_object.rs

```diff
@@ -0,0 +1,122 @@
+use super::TypedArrayName;
+use crate::{object::JsObject, JsResult};
+
+/// Type of the array content.
+#[derive(Debug, Clone, Copy)]
+pub(super) enum ContentType {
+    Number,
+    BigInt,
+}
+
+/// <https://tc39.es/ecma262/#integer-indexed-exotic-object>
+#[derive(Debug, Clone)]
+pub struct IntegerIndexedObject {
+    pub(super) prototype: JsObject,
```

Comment:
Why `IntegerIndexedObject` has to hold its own prototype? Wouldn't this make `IntegerIndexedObjects` have two prototypes when we create a new `JsObject` with kind `IntegerIndexed`?

Reply:
Indeed, I was just playing around following the spec too closely xD